API.md

```diff
@@ -220,7 +220,8 @@ polledExpect(() => myValue).toBe(true);
 * **polledExpect(func, waitTimeMs?)** - Works like jasmine's "expect", but retries the function until it passes or times out.
 * **resizeViewport(size: { width?: number; height?: number; })** - Resize the viewport (not the window) to the specified size.
 * **takeScreenshot(filename)** - Takes a screenshot and saves a .png file at the specified file path.
-* **waitFor(condition, waitTimeMS?)** - Waits for the condition function to return a truthy value. An exception will be raised if it times out.
+* **waitFor(condition: () => boolean, waitTimeMS?)** - Waits for the condition function to return a truthy value. An exception will be raised if it times out.
+* **waitFor<T>(condition: () => { data: T, keepPolling: boolean })** - Waits fo the condition function to return { keepPolling: false } and returns the data element. An exception will be raised if it times out.
```

Comment:
I dig the flexibility of this, thanks @scriby !

Reply:
@here: Thoughts about making this a 5.2.0 version as this seems to be adding new functionality?